dragonfly/implementations/at_operator.py

```diff
@@ -0,0 +1,460 @@
+
+    def parse_output(self, rtm_output, bot_id):
+        '''
+        Parse the given Slack output and check whether or not I am called.
+        rtm_output: Slack runtime output
+        bot_id    : my bot id
+        '''
+        output = rtm_output
+        if output and len(output) > 0:
+            for o in output:
+                if o and 'text' in o:
+                    if ('@' + str(bot_id)) in o['text']:
+                        return o['channel']
+                    tokens = o['text'].split(' ', 1)
+                    command = tokens[0]
+                    if command in self.command_dictionary:
+                        logger.info(' A helper command [' + command + '] is called!')
+                        channel = o["channel"]
+                        user_id = o["user"]
+                        operator_name = ""
+                        if len(tokens) > 1 and tokens[1].startswith('[') and "]" in tokens[1]:
+                            sub_tokens = tokens[1].split(']', 1)
+                            operator_name = sub_tokens[0].replace('[', '')
+                            logger.info(' Found a potential operator name [' + operator_name + '] !')
+                        func = self.command_dictionary[command]
+                        num_args = len(funcsigs.signature(func).parameters)
+                        args = [channel, user_id, operator_name]
+                        func(*args[:num_args])
+        return None
+
+    def run(self):
+        '''
+        The main loop.
+        '''
+        logging.getLogger('googleapiclient.discovery_cache').setLevel(logging.ERROR)
+        self.get_slack_client()
+        if self.slack_client.rtm_connect(auto_reconnect=True):
+            logger.info(' Connected!')
+            bot_id = self.slack_client.api_call('auth.test')['user_id']
+            if not bot_id:
+                logger.critical(' Unable to get the bot user ID.')
+                os._exit(1)
+            logger.info(' Got Slack Operator bot id.')
+
+            self.full_name_to_id_dictionary, self.id_to_username_dictionary, self.username_to_id_dictionary = self.construct_user_dictionaries()
+            if not self.full_name_to_id_dictionary:
+                logger.critical(" Cannot get the list of users.")
+                os._exit(1)
+
+            self.channel_name_to_id_dictionary, self.channel_id_to_name_dictionary = self.construct_channel_dictionaries()
+            if not self.channel_name_to_id_dictionary:
+                logger.error(' Cannot get the list of channels, therefore cannot find the monitor channel. Related functionalities will be disabled.')
+            elif self.monitor_channel_name not in self.channel_name_to_id_dictionary:
+                logger.error(' The name of monitor channel is not found. Related functionalities will be disabled.')
+            else:
+                self.monitor_channel_id = self.channel_name_to_id_dictionary[self.monitor_channel_name]
```

Comment:
the `monitor_channel_id` variable is a blank string, the logger from the next line indicated the coded did reach this else

Reply:
By the time it reaches the else branch the `monitor_channel_id` is supposed to be a blank string - it is not assigned any value until line 393.